src/Scientist/Observation.cs

```diff
@@ -1,49 +1,178 @@
-﻿using GitHub.Internals;
+﻿using NullGuard;
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace GitHub
 {
     /// <summary>
-    /// Defines an observation of an experiment's execution.
+    /// Provides an interface for an observation of an experiment's execution.
     /// </summary>
-    public class Observation
+    public interface IObservation
```

Comment:
I don't think we need to introduce an interface here just yet. This type is pretty much just a DTO and it's easy to instantiate.